### PR TITLE
gradients

### PR DESCRIPTION
**Pull request by KartikeyDubeyKD**

### CODE REVIEW

**Comment by KartikeyDubeyKD** [comment]
#14818

**Comment by KartikeyDubeyKD** [comment]
@CatB1t @YushaArif99  I think there are some issues with general_functions testing, I'm trying to run tests locally it always gives the `ERROR: not found`

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py [hunk 1]
```diff
@@ -234,6 +235,43 @@ def get_shape(
...
+        input_dtypes = input_dtypes or (tf.float32,)
+        output_grad_shapes = output_grad_shapes or input_shapes
+        output_grad_dtypes = output_grad_dtypes or input_dtypes
+
+        inputs = tuple(tf.ones(shape=shape, dtype=dtype) for shape, dtype in zip(input_shapes, input_dtypes))
+        output_tensors = tf.gradients(tf.reduce_sum(tf.stack([tf.reduce_prod(inp) for inp in inputs])), inputs)
+
+        output_grads = tuple(tf.ones(shape=shape, dtype=dtype) for shape, dtype in zip(output_grad_shapes, output_grad_dtypes))
+
+        return inputs, output_grads
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:272]: We shouldn't have framework or frontend specific strategies, instead we should use Hypothesis and Ivy strategies to generate valid arguments.

**Review by CatB1t — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py (hunk 1 above)
Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:272]: ok sir I got that but,  think there are some issues with general_functions testing, I'm trying to run tests locally it always gives the `ERROR: not found`

Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:272]: image

Comment (YushaArif99) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:272]: Hi @KartikeyDubeyKD . I think the issue is most likely related to an incorrect path in your pytest configuration for the test you're running. It's best to follow the guide regarding setting up testing as mentioned in the Docs [here](https://unify.ai/docs/ivy/overview/contributing/setting_up.html#setting-up-testing-in-pycharm)

Comment (KartikeyDubeyKD) [reply ivy_tests/test_ivy/helpers/hypothesis_helpers/general_helpers.py:272]: No @YushaArif99 other functions are running fine, I'm just facing a problem in this one only, can you please check once

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.
